contracts/VotingMachines/Governance.sol

```diff
@@ -0,0 +1,509 @@
+        });
+        return hashedParameters;
+    }
+
+  /**
+   * @dev hashParameters returns a hash of the given parameters
+   */
+    function getParametersHash(
+        Reputation _reputationSystem, // the reputation system that is being used
+        uint _noneBoostedVoteRequierePercentage, // the absolute vote percentages bar.
+        uint _noneBoostedVotePeriodLimit, //the time limit for a proposal to be in an absolute voting mode.
+        uint _boostedVotePeriodLimit, //the time limit for a proposal to be in an relative voting mode.
+        uint _scoreThreshold,
+        GovernanceFormulasInterface _governanceFormulasInterface,
+        uint _minimumStaketingFee) public pure returns(bytes32)
+        {
+        return keccak256(
+            _reputationSystem,
+            _noneBoostedVoteRequierePercentage,
+            _noneBoostedVotePeriodLimit, //the time limit for a proposal to be in an absolute voting mode.
+            _boostedVotePeriodLimit, //the time limit for a proposal to be in an relative voting mode.
+            _scoreThreshold,
+            _governanceFormulasInterface,
+            _minimumStaketingFee);
+    }
+
+  /**
+   * @dev register a new proposal with the given parameters. Every proposal has a unique ID which is being
+   * generated by calculating keccak256 of a incremented counter.
+   * @param _paramsHash defined the parameters of the voting machine used for this proposal
+   * @param _avatar an address to be sent as the payload to the _executable contract.
+   * @param _executable This contract will be executed when vote is over.
+   */
+    function propose(uint _numOfChoices, bytes32 _paramsHash, address _avatar, ExecutableInterface _executable) public returns(bytes32) {
+        // Check valid params and number of choices:
+        require(parameters[_paramsHash].reputationSystem != address(0));
+        require(_numOfChoices > 0 && _numOfChoices <= MAX_NUM_OF_CHOICES);
+        require(ExecutableInterface(_executable) != address(0));
+        // Generate a unique ID:
+        bytes32 proposalId = keccak256(this, proposalsCnt);
+        proposalsCnt++;
+        // Open proposal:
+        Proposal memory proposal;
+        proposal.numOfChoices = _numOfChoices;
+        proposal.avatar = _avatar;
+        proposal.executable = _executable;
+        proposal.state = ProposalState.NoneBoosted;
+        // solium-disable-next-line security/no-block-members
+        proposal.submittedTime = now;
+        proposals[proposalId] = proposal;
+        NewProposal(proposalId, _numOfChoices, msg.sender, _paramsHash);
+        return proposalId;
+    }
+
+  /**
+   * @dev Cancel a proposal, only the owner can call this function and only if allowOwner flag is true.
+   * @param _proposalId the proposal ID
+   */
+    function cancelProposal(bytes32 _proposalId) public onlyProposalOwner(_proposalId) votable(_proposalId) returns(bool) {
+        //This is not allowed.
+        return false;
```

Comment:
Should this revert instead of returning false, or perhaps add an `allowed(false)` modifier?

Reply:
cancelProposal is not allowed for this votingMachine ( yet).
It is there just because of the voting interface.
Not sure it should revert. 
One should not call that and if it does it will just do nothing.